rusk-recovery/src/state.rs

```diff
@@ -67,124 +41,15 @@ fn diskbackend() -> BackendCtor<DiskBackend> {
-    }
-
-    transfer
-        .update_root()
-        .expect("Root to be updated after pushing genesis note");
-
-    let stake_amount = stake_amount(testnet);
-    let stake_balance = stake_amount * PROVISIONERS.len() as u64;
-
-    transfer
-        .add_balance(rusk_abi::stake_contract(), stake_balance)
-        .expect("Stake contract balance to be set with provisioner stakes");
-
-    transfer
-}
-
-const fn stake_amount(testnet: bool) -> Dusk {
-    match testnet {
-        true => dusk(2_000_000.0),
-        false => MINIMUM_STAKE,
-    }
-}
-
-/// Creates a new stake contract state with preset stakes added for the
-/// staking/consensus keys in the `keys/` folder. The stakes will all be the
-/// same and the minimum amount.
-fn genesis_stake(testnet: bool) -> StakeContract {
-    let theme = Theme::default();
-    let mut stake_contract = StakeContract::default();
-
-    let stake_amount = stake_amount(testnet);
-
-    for provisioner in PROVISIONERS.iter() {
-        let stake = Stake::with_eligibility(stake_amount, 0, 0);
-        stake_contract
-            .insert_stake(*provisioner, stake)
-            .expect("Genesis stake to be pushed to the stake");
-    }
-    info!(
-        "{} Added {} provisioners",
-        theme.action("Generating"),
-        PROVISIONERS.len()
-    );
-
-    stake_contract
-}
-
+#[cfg(not(feature = "genesis"))]
 pub fn deploy<B>(
-    testnet: bool,
-    ctor: &BackendCtor<B>,
+    _: bool,
+    _: &BackendCtor<B>,
 ) -> Result<NetworkStateId, Box<dyn Error>>
 where
-    B: 'static + Backend,
+    B: 'static + microkelvin::Backend,
 {
-    Persistence::with_backend(ctor, |_| Ok(()))?;
-
-    let theme = Theme::default();
-    info!("{} new network state", theme.action("Generating"));
-
-    let transfer = Contract::new(
-        genesis_transfer(testnet),
-        &include_bytes!(
-      "../../target/wasm32-unknown-unknown/release/transfer_contract.wasm"
-    )[..],
-    );
-
-    let stake = Contract::new(
-        genesis_stake(testnet),
-        &include_bytes!(
-            "../../target/wasm32-unknown-unknown/release/stake_contract.wasm"
-        )[..],
-    );
-
-    let mut network = NetworkState::default();
-
-    info!(
-        "{} Genesis Transfer Contract state",
-        theme.action("Deploying")
-    );
-
-    network
-        .deploy_with_id(rusk_abi::transfer_contract(), transfer)
-        .expect("Genesis Transfer Contract should be deployed");
-
-    info!("{} Genesis Stake Contract state", theme.action("Deploying"));
-
-    network
-        .deploy_with_id(rusk_abi::stake_contract(), stake)
-        .expect("Genesis Transfer Contract should be deployed");
-
-    info!("{} network state", theme.action("Storing"));
-
-    network.commit();
-    network.push();
-
-    info!("{} {}", theme.action("Root"), hex::encode(network.root()));
-
-    let state_id = network.persist(ctor).expect("Error in persistence");
-
-    Ok(state_id)
+    Err("No 'genesis' feature".into())
```

Comment:
If the function can't be called, why include it at all?

Reply:
To help the user whom doesn't build it with the feature enable to figure out the problem.
Removing completely lead to removal of `-w` flag too, and user will face just a "unknown argument" error with no clue about the root cause